### PR TITLE
DMP-4228 ARM RPO Update fields 

### PR DESCRIPTION
**Pull request by DavidTMann**
https://tools.hmcts.net/jira/browse/DMP-4228

Amended edit batch size component to edit automated task component, to handle batch/datetime amendments.

Validation refactored to utilise forms service

Task dates are stringified before passing through state as object type does not persist, this allows refresh on edit screen without breaking functionality.

Validation messages for datepicker and time picker have been assumed as not mentioned in ticket.

Request to update date values simply sends key we receive, with new value e.g. { rpo_csv_start_hour: "2023-05-15T13:30:00.000Z" }

Unit tests/cypress tests added

Values are pre-populated in form if they exist

Change links appear for empty values if attribute is set

Screenshot 2024-11-13 at 16 35 05

### CODE REVIEW

**Review by davet1985 — COMMENTED** [review]
A few minor comments, looks good.

File: src/app/admin/components/automated-tasks/edit-automated-task/edit-automated-task.component.ts [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+      case 'ARM_START_TIME':
+        return 'arm_replay_start_ts';
+      case 'ARM_END_TIME':
+        return 'arm_replay_end_ts';
+      default:
+        return null;
+    }
+  }
+
+  private getUTCISODateString(): string | null {
```
[the remark this answers is not in the document]
Comment (DavidTMann) [reply src/app/admin/components/automated-tasks/edit-automated-task/edit-automated-task.component.ts:238]: Had a brief look and made some adjustments to the function so we do the actual UTC & ISO conversion in the service. I think it's worth raising a separate ticket to see how we can standardise this scenario across the app, there seems to be different approaches but it'd be nice to get these all in line.

[the remark this answers is not in the document]
Comment (davet1985) [reply src/app/admin/components/automated-tasks/edit-automated-task/edit-automated-task.component.ts]: Ok, sounds good, please create a ticket for this.

**Review by davet1985 — APPROVED** [review]
🦫

**Pull request merged**